plugins/retry/errors.js

```diff
@@ -29,7 +29,9 @@ class RetryError extends MessageError {
 }
 
 function isRetryable(err) {
-    if (RetryError.promotable(err)) return false;
+    if (err instanceof RetryError ||
+        RetryError.promotable(err) ||
+        err.cause && err.cause.message.startsWith('Channel ended')) return false;
```

Comment:
Is there any possibility to be based on error codes rather than this string?

It does not matter much but I wonder if the message string can be different based on the broker used

Reply:
No this one is from the client library. We are not yet handling server replies.

Reply:
And it is what `TODO` implies below.

Reply:
ok got it thanks